<robot xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
       xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
       xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface"
       xmlns:xacro="http://ros.org/wiki/xacro"
	   name="pan_tilt_adapter"
	>
	
		 <link name="pantilt_base_link">	
			<visual> 	  	
				<geometry>					
			  		<mesh filename="package://h4r_pantilt_adapter/meshes/basepart.dae"  />
				</geometry>
			</visual>

			<collision name="pantilt_foot_link_collision">
				<geometry>	
					<mesh filename="package://h4r_pantilt_adapter/meshes/basepart.dae"  />
				</geometry>	
			</collision>

			<inertial>
				<mass value="0.010" />
				<inertia ixx="3.8" 
						 iyy="3.8" 
						 izz="7.6" 
				         iyz="0.0" 
				         ixy="0.0" 
				         ixz="0.0" />
			</inertial>
		</link>	

		
		
	<joint name="pantilt_pan_joint" type="revolute">
      <axis xyz="0 0 1"/>
      <limit lower="0" upper="1" effort="9" velocity="1" />      
      <origin xyz="0 0 0.0028" rpy="0 0 0" />
      <parent link="pantilt_base_link"/>
      <child link="pantilt_center_link"/>
    </joint>
		
		
		
		
		<link name="pantilt_center_link">
			<visual> 	  	
				<geometry>					
			  		<mesh filename="package://h4r_pantilt_adapter/meshes/centerpart.dae"  />
				</geometry>
			</visual>

			<collision name="pantilt_center_link_collision">
				<geometry>	
					<mesh filename="package://h4r_pantilt_adapter/meshes/centerpart.dae"  />
				</geometry>	
			</collision>

			<inertial>
				<mass value="0.025" />
				<inertia ixx="9.325" 
						 iyy="11.911" 
						 izz="7.028" 
				         iyz="0.0" 
				         ixy="0.0" 
				         ixz="1.025" />
			</inertial>
		</link>	
		
		
	<joint name="pantilt_tilt_joint" type="revolute">
      <axis xyz="1 0 0"/>
      <limit lower="0" upper="1" effort="9" velocity="1" />
      <origin xyz="-0.022267 0 0.045495" rpy="0 0 3.14" />
      <parent link="pantilt_center_link"/>
      <child link="pantilt_top_link"/>
    </joint>
		
		
		<link name="pantilt_top_link">	
			
			<visual> 	  	
				<geometry>					
			  		<mesh filename="package://h4r_pantilt_adapter/meshes/toppart.dae"  />
				</geometry>
			</visual>

			<collision name="pantilt_top_link_collision">
				<geometry>	
					<mesh filename="package://h4r_pantilt_adapter/meshes/toppart.dae"  />
				</geometry>	
			</collision>

			<inertial>
				<mass value="0.010" />
				<inertia ixx="4.481" 
						 iyy="2.930" 
						 izz="5.819" 
				         iyz="-0.083" 
				         ixy="0.049" 
				         ixz="0.317" />
			</inertial>
		</link>	
		
</robot>
